File: app/src/main/java/com/rcdvl/marvel/resourcedetails/ResourceDetailsPageFragment.java
package com.rcdvl.marvel.resourcedetails;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.rcdvl.marvel.R;
import com.rcdvl.marvel.model.api.MarvelResource;

import org.parceler.Parcels;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by renan on 04/07/16.
 */
public class ResourceDetailsPageFragment extends Fragment {

    public static String EXTRA_RESOURCE = "extra-resource";
    private Unbinder unbinder;
    @BindView(R.id.resourceText) TextView resourceText;
    @BindView(R.id.resourceImage) ImageView resourceImage;

    public static Fragment newInstance(MarvelResource resource) {
        ResourceDetailsPageFragment fragment = new ResourceDetailsPageFragment();
        Bundle args = new Bundle();
        args.putParcelable(EXTRA_RESOURCE, Parcels.wrap(resource));
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_resource_page, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        unbinder = ButterKnife.bind(this, view);
        MarvelResource resource = Parcels.unwrap(getArguments().getParcelable(EXTRA_RESOURCE));
        Glide.with(getContext()).load(resource.thumbnail.path + "." + resource.thumbnail.extension).crossFade().into(resourceImage);
        resourceText.setText(resource.title);
    }

    @Override
    public void onDestroyView() {
        unbinder.unbind();
        super.onDestroyView();
    }
}

File: app/src/main/kotlin/com/rcdvl/marvel/resourcedetails/ResourceDetailsPageFragment.kt
package com.rcdvl.marvel.resourcedetails

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.rcdvl.marvel.R
import com.rcdvl.marvel.networking.MarvelResource
import kotlinx.android.synthetic.main.fragment_resource_page.*

/**
 * Created by renan on 3/21/16.
 */
class ResourceDetailsPageFragment : Fragment() {

    companion object {
        fun newInstance(resource: MarvelResource): ResourceDetailsPageFragment {
            val fragment: ResourceDetailsPageFragment = ResourceDetailsPageFragment()
            var args = Bundle()
            args.putSerializable(EXTRA_RESOURCE, resource)
            fragment.arguments = args
            return fragment
        }

        const val EXTRA_RESOURCE = "extra-resource"
        const val EXTRA_SOURCE = "extra-source"
        const val EXTRA_RESOURCE_TYPE = "extra-resource-type"
    }


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(R.layout.fragment_resource_page, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        var resource = arguments.getSerializable(EXTRA_RESOURCE)

        if (resource != null) {
            resource = resource as MarvelResource

            Glide.with(context)
                    .load(resource.thumbnail?.path + '.' + resource.thumbnail?.extension)
                    .crossFade()
                    .into(resourceImage)

            resourceText.text = resource.title
        }
    }
}